### PR TITLE
feat(devtools): useGotoAtomsSnapshot

### PR DESCRIPTION
**Pull request by Thisen**
Adds a hook to go to a new snapshot, replacing state in the store.

### CODE REVIEW

File: src/devtools/useGoToAtomsSnapshot.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  if (!isDevStore(store)) {
+    throw Error('useGoToAtomsSnapshot can only be used in dev mode.')
+  }
+  const restoreAtoms = store[3]
+  return useCallback<(...args: Parameters<typeof restoreAtoms>) => void>(
+    (...args) => {
+      restoreAtoms(...args)
+    },
+    [restoreAtoms]
+  )
```
Comment (dai-shi) [inline src/devtools/useGoToAtomsSnapshot.ts:20]:
```suggestion
  return restoreAtoms
```
Does this work?

File: src/core/Provider.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ const stateToPrintable = ([state, atoms]: [State, AnyAtom[]]) =>
     })
   )
 
-const isDevStore = (store: Store): store is StoreForDevelopment => {
+export const isDevStore = (store: Store): store is StoreForDevelopment => {
   return store.length > 2
 }
```
Comment (dai-shi) [inline src/core/Provider.ts:62]: Maybe we want to move this to contexts.ts?

File: tests/devtools/useGoToAtomsSnapshot.test.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { fireEvent, render } from '@testing-library/react'
+import { Provider, atom, useAtom } from '../../src/index'
+import { useAtomsSnapshot, useGoToAtomsSnapshot } from '../../src/devtools'
+
+beforeEach(() => {
+  process.env.NODE_ENV = 'development'
+})
+afterEach(() => {
+  process.env.NODE_ENV = 'test'
+})
```
Comment (dai-shi) [inline tests/devtools/useGoToAtomsSnapshot.test.tsx:11]:
```suggestion
```

We probably don't need this any more (if you merge master).

File: src/core/contexts.ts [hunk 4]
```diff
@@ -25,7 +25,8 @@ export type StoreForDevelopment = [
     atoms: AnyAtom[]
     state: State
     listeners: Set<() => void>
-  }>
+  }>,
+  restoreAtoms: (values: Iterable<readonly [AnyAtom, unknown]>) => void
```
Comment (dai-shi) [inline src/core/contexts.ts:29]:
```suggestion
  restore: (values: Iterable<readonly [AnyAtom, unknown]>) => void
```
just to be consistent.

File: src/devtools/useAtomsSnapshot.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { useContext, useMemo } from 'react'
 import { SECRET_INTERNAL_getStoreContext as getStoreContext } from 'jotai'
-import type { AnyAtom } from '../core/types'
+import type { AnyAtom, Scope } from '../core/types'
```
Comment (dai-shi) [inline src/devtools/useAtomsSnapshot.ts:3]:
```suggestion
import type { AnyAtom } from '../core/types'
```
It's not used, right?

**Review by dai-shi — APPROVED** [review]
Looks great. Nice work!!

File: src/devtools/useGotoAtomsSnapshot.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+export function useGotoAtomsSnapshot(scope?: Scope) {
+  const StoreContext = getStoreContext(scope)
+  const store = useContext(StoreContext)
+
+  if (!isDevStore(store)) {
+    throw Error('useGotoAtomsSnapshot can only be used in dev mode.')
+  }
+  const restoreAtoms = store[3]
+  return restoreAtoms
```
Comment (dai-shi) [inline src/devtools/useGotoAtomsSnapshot.ts:15]: How about checking the misuse of `scope`?

```suggestion
  return useCallback((values: Parameters<typeof restore>[0]) => {
    for (const [atom] of values) {
      if (atom.scope !== scope) {
        throw new Error('atom scope mismatch to restore')
      }
    }
    restore(values)
  }, [restore]
```

Too much? (I renamed `restore` from `restoreAtoms`.)

Comment (Thisen) [reply src/devtools/useGotoAtomsSnapshot.ts:15]: I should probably write some tests aswell?

**Pull request merged**